### ISSUE TITLE
New Rule: `no-else`

### ISSUE THREAD

**Issue by raphael-arce** [eslint/eslint#19871]
### Rule details

An ESLint plugin that forbids else blocks and encourages early returns instead

### Related ECMAScript feature

None

### What type of rule is this?

Suggests an alternate way of doing something

### Example code

```js
/**
 * Bad example
 * /
function getSign(x) {
    if (x > 0) {
      return '+';
    } else if (x < 0) {
      return '-';
    } else {
      return '';
    }
}

/**
 * Good example
 * /
function getSign(x) {
  if (x > 0) {
    return '+';
  }

  if (x < 0) {
    return '-';
  }

  return '';
}
```

### Why should this rule be in the core instead of a plugin?

This makes the code flow more linear and easier to follow / understand. 

### Participation

- [x] I am willing to submit a pull request to implement this rule.

### Additional comments

Here would be an example implementation of the rule: https://github.com/raphael-arce/eslint-plugin-no-else/blob/main/src/no-else.js 
(First time writing an EsLint plugin/rule, not sure the implementation is correct)

**Comment by mdjermanovic** [comment]
Hi @raphael-arce, thanks for the rule proposal!

We already have a core rule that disallows `else` after `return`: https://eslint.org/docs/latest/rules/no-else-return. Here's a [Playground demo](https://eslint.org/play/#eyJ0ZXh0IjoiLyoqXG4gKiBCYWQgZXhhbXBsZVxuICovXG5mdW5jdGlvbiBnZXRTaWduKHgpIHtcbiAgICBpZiAoeCA+IDApIHtcbiAgICAgIHJldHVybiAnKyc7XG4gICAgfSBlbHNlIGlmICh4IDwgMCkge1xuICAgICAgcmV0dXJuICctJztcbiAgICB9IGVsc2Uge1xuICAgICAgcmV0dXJuICcnO1xuICAgIH1cbn1cblxuLyoqXG4gKiBHb29kIGV4YW1wbGVcbiAqL1xuZnVuY3Rpb24gZ2V0U2lnbjIoeCkge1xuICBpZiAoeCA+IDApIHtcbiAgICByZXR1cm4gJysnO1xuICB9XG5cbiAgaWYgKHggPCAwKSB7XG4gICAgcmV0dXJuICctJztcbiAgfVxuXG4gIHJldHVybiAnJztcbn0iLCJvcHRpb25zIjp7InJ1bGVzIjp7Im5vLWVsc2UtcmV0dXJuIjpbImVycm9yIix7ImFsbG93RWxzZUlmIjpmYWxzZX1dfSwibGFuZ3VhZ2VPcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fX19fX0=) with the example code from the original post.

If you want to entirely disable `else`, it can be easily done by configuring the [`no-restricted-syntax`](https://eslint.org/docs/latest/rules/no-restricted-syntax) rule with the following selector:

```
IfStatement > *.alternate
```

Here's a [Playground demo](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLXJlc3RyaWN0ZWQtc3ludGF4OiBbXCJlcnJvclwiLCB7XG4gIHNlbGVjdG9yOiBcIklmU3RhdGVtZW50ID4gKi5hbHRlcm5hdGVcIixcbiAgbWVzc2FnZTogXCJEb24ndCB1c2UgZWxzZVwiXG59XSAqL1xuXG5pZiAoZm9vKSB7XG4gIGJhcigpO1xufSBlbHNlIHtcbiAgYmF6KCk7XG59Iiwib3B0aW9ucyI6eyJydWxlcyI6eyJuby1lbHNlLXJldHVybiI6WyJlcnJvciIseyJhbGxvd0Vsc2VJZiI6ZmFsc2V9XX0sImxhbmd1YWdlT3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e319fX19) with the `no-restricted-syntax` rule.

Per our [rule policies](https://eslint.org/blog/2020/05/changes-to-rules-policies/), we are not accepting new rules that only disallow syntax, so this rule doesn't qualify for inclusion in the core. If existing solutions don't work for you, you can always make a custom rule and publish a plugin.
